### PR TITLE
[rigor] Decouple num_trials from library size — a strategy's DSR depends only on itself (decouple #2)

### PR DESCRIPTION
**Pull request by dbrowneup**
## Principle (Dan, 2026-07-09)
**A strategy's statistical rigor must depend ONLY on that strategy — never on how many other strategies (curated or generated) sit in the library.** Today the DSR multiple-testing `num_trials` keys off the curated library COUNT, so adding/removing a fixture silently changes every other strategy's Deflated Sharpe. This is the exact violation from the decouple+consolidate plan (`docs/CURATED-STRATEGY-DECOUPLE-AND-CONSOLIDATE-2026-07-08.md` Part A #2).

## What changed — `num_trials` is now self-contained everywhere
- **Generated (society/debate) path** → `num_trials` = the strategy's OWN candidate pool `N` (`_society_num_trials(pool)`), dropping the `+ library_size` term. `generation_pipeline.py`, `debate_engine.py`.
- **Curated `/gate` + batch/live paths** → `num_trials = 1` (each curated single-paper strategy graded on its own Sharpe; with `num_trials=1` the DSR expectation-of-max term collapses). `selection_bias_routes.py`, `live_rigor_gate.py` (`_default_num_trials`, `verdicts_for_strategies`), `strategies_routes.py` (`_live_rigor_results_for_strategies`, `_run_fusion_job`).
- **Fusion default** → `fusion_evaluator._default_num_trials()` returns `1` (was library size). A strategy's own parameter-variant grid stays a legitimate self-contained selection layer (the `max(num_trials, len(variants))` logic is untouched).
- Removed dead `_library_size()`; reworded all stale "N + library" / #770/#820 rationale.

## ⚠️ Needs Dan + Önder sign-off before merge
1. **This REVERSES the `N + library_size` convention from #770/#811/#820** — Önder argued the library is a second selection layer. Dan's principle is that it isn't (a strategy's rigor can't change because the library grew).
2. **It raises rigor pass rates** — removing the cross-strategy penalty means more strategies clear the DSR gate. That's the *correct* consequence of the fix, but it's a visible, claim-integrity-adjacent shift, so it should land deliberately, not silently.

## What did NOT change (verified)
- **No rigor FLOOR value moved** (DSR_P_FLOOR, OOS, look-ahead — only stale rationale comments reworded; `dsr_p_min=0.90` untouched, flagged for Önder's re-derivation given the deflation denominator changed).
- **PBO and average_correlation semantics untouched** — they stay cohort-derived; only `num_trials` was decoupled.

## Tests
- Tests that encoded the old `N + library` / library-size math updated to the new self-contained semantics (not weakened).
- Fixed a **real latent test bug** exposed once `num_trials` stopped dominating the verdict (`test_library_badge_equals_live_gate_verdict_on_persisted_returns` was missing a `_load_strategy_code_safe_local` monkeypatch → the look-ahead leg read uncontrolled source).
- ✅ 617 rigor-relevant tests pass; full `-k "not integration"` sweep green (2843); ruff clean. Independently re-verified by the main session.

Part of the curated decouple+consolidate program. Follow-ups (separate PRs): the rebalancer decouple (#1, generated vaults never rebalance), and the 34→~6 consolidation (Part B).

### CODE REVIEW

**Comment by onder-akkaya** [comment]
Reviewed. Notes below.

**CI status:** Quality Gate, Complexity Gate, and import-guard all show red on the PR, but it's not a code failure — every job in those runs has the annotation "The job was not acquired by Runner of type hosted even after multiple attempts." That's a runner-availability problem, not a test/lint failure. Worth a re-run once hosted runners are picking up jobs again before merging on the strength of a green check. Copilot's automated review run did complete successfully.

**Correctness:** Traced every callsite of `_society_num_trials` and the `_default_num_trials` helpers across `generation_pipeline.py`, `debate_engine.py`, `strategies_routes.py`, `selection_bias_routes.py`, `live_rigor_gate.py`, and `fusion_evaluator.py`. All are consistently converted to the single-arg / self-contained-1 convention, no leftover two-arg calls, no place where `library_size` still leaks into `num_trials`. The `_library_size()` helper in `debate_engine.py` is fully removed rather than left dead. This does what the title says.

One stale comment: `fusion_evaluator.py`, in `apply_rigor_gate`, the block right above `effective_trials = max(num_trials, len(variants_metrics))` still reads "on top of whatever selection set the caller already accounted for (society N + library, #820, or the plain library-size fallback...)". That's describing the old pre-decouple convention as if it's still how callers populate `num_trials`. The `max()` logic itself is fine and untouched, but the comment should get scrubbed in the same pass that touched everything else in that file — as is it contradicts `_default_num_trials()`'s own new docstring three lines up.

**Review comments:** Both of Copilot's comments are still open (from ~13:37 UTC today, no reply/resolution visible):
- `test_generation_pipeline.py` — `test_independent_of_library_size` is tautological (`_society_num_trials(20) == _society_num_trials(20)` always holds regardless of the fix). Fair catch, the test doesn't actually pin down library-independence.
- `rigor_evaluator.py` — the `num_trials == 1` warning log will now fire on basically every curated/single-candidate gate run since 1 is the new expected default, not an edge case. Worth downgrading to debug or gating it so it doesn't turn into permanent log noise.

Both are legitimate, neither blocks correctness of the core change.

File: backend/tests/test_generation_pipeline.py [hunk 1]
```diff
@@ -80,40 +80,47 @@ def test_failing_gate_is_never_deployable(self):
...
+        # N=1 (one generated candidate) is a single-trial grade — no library term added.
+        assert _society_num_trials(1) == 1
 
     def test_floored_at_one(self):
-        assert _society_num_trials(library_size=0, selection_pool_size=0) == 1
+        assert _society_num_trials(0) == 1
+
+    def test_independent_of_library_size(self):
+        # The whole point: growing the library must NOT change a strategy's num_trials.
+        assert _society_num_trials(20) == _society_num_trials(20)
```
[the remark this answers is not in the document]
Comment (dbrowneup) [reply backend/tests/test_generation_pipeline.py:109]: Fixed in b4481b80 — the test now pins library-independence structurally: a single-parameter signature assertion plus `pytest.raises(TypeError)` on the old `(N, library_size)` call shape.

File: backend/archimedes/services/rigor_evaluator.py [hunk 2]
```diff
@@ -815,13 +815,20 @@ def run_rigor_gate(
...
+        # Loud on purpose (#902, reworded for decouple #2): num_trials=1 zeroes
+        # E[max SR] and collapses DSR to a plain "Sharpe > 0" test. This is the
+        # CORRECT, expected value for a strategy graded on its own self-contained
+        # selection set (a curated single-paper strategy, or a strategy with no
+        # real N-candidate search behind it) — a strategy's rigor depends ONLY on
+        # itself, never on the count of other strategies in the library (Dan's
+        # principle). The warning stays as a diagnostic breadcrumb: if a caller
+        # meant to pass the strategy's OWN selection-pool size (its N generated
+        # candidates or parameter-variant grid) and forgot, this line is the tell.
         logger.warning(
```
[the remark this answers is not in the document]
Comment (dbrowneup) [reply backend/archimedes/services/rigor_evaluator.py]: Fixed in b4481b80 — demoted to `logger.debug` with a comment explaining the change: post-decouple, num_trials=1 is the documented default on every curated serving call, so WARNING was hot-path spam; debug keeps the misuse breadcrumb.

File: docs/specs/selection-bias-corrections-spec.md [hunk 3]
```diff
@@ -156,6 +156,40 @@ fresh candidate pool, so that route keeps `num_trials = library_size`.
 > statistics call. This addendum records approach **A** as the shipped default pending his
 > review; promoting to B is a follow-up if candidate-pool over-deflation proves material.
 
+#### Addendum (#1075, 2026-07-14) — decouple #2: num_trials is self-contained — SUPERSEDES the `+ library_size` term above
+
+PR #1075 deliberately reverses formula (A)'s `+ library_size` term. This is the one
+documented exception to #770's "only stricter" anti-goal, reviewed as such — the
+loosening is the point, not a side effect. Owner + statistics sign-off is the merge
+of that PR itself.
```
[the remark this answers is not in the document]
Comment (dbrowneup) [reply docs/specs/selection-bias-corrections-spec.md:167]: Fixed in 3996d992 — the #770 scope note now carries an explicit supersession marker (including for `/api/selection-bias/gate`, which this PR moves to num_trials=1), so no reader takes the stale paragraph as current.
